app/controllers/charityortrust/add/charity/CheckDetailsController.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.charityortrust.add.charity
+
+import config.FrontendAppConfig
+import connectors.TrustConnector
+import controllers.actions._
+import controllers.actions.charity.NameRequiredAction
+import javax.inject.Inject
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.PlaybackRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import utils.mappers.CharityBeneficiaryMapper
+import utils.print.CharityBeneficiaryPrintHelper
+import viewmodels.AnswerSection
+import views.html.charityortrust.add.charity.CheckDetailsView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CheckDetailsController @Inject()(
+                                        override val messagesApi: MessagesApi,
+                                        standardActionSets: StandardActionSets,
+                                        val controllerComponents: MessagesControllerComponents,
+                                        view: CheckDetailsView,
+                                        connector: TrustConnector,
+                                        val appConfig: FrontendAppConfig,
+                                        playbackRepository: PlaybackRepository,
+                                        printHelper: CharityBeneficiaryPrintHelper,
+                                        mapper: CharityBeneficiaryMapper,
+                                        nameAction: NameRequiredAction
+                                      )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  def onPageLoad(): Action[AnyContent] = standardActionSets.verifiedForUtr.andThen(nameAction) {
+    implicit request =>
+
+      val section: AnswerSection = printHelper(request.userAnswers, request.beneficiaryName)
+      Ok(view(section))
+  }
+
+  def onSubmit(): Action[AnyContent] = standardActionSets.verifiedForUtr.async {
+    implicit request =>
+
+      mapper(request.userAnswers) match {
+        case None =>
+          Future.successful(InternalServerError)
```

Comment:
@jackheslop96 This won't render the Tech difficulties page.

I've done this in amend individual beneficiary:

```
import config.ErrorHandler

@Inject()(errorHandler: ErrorHandler.)

Future.successful(InternalServerError(errorHandler.internalServerErrorTemplate))
```

Reply:
Thanks Adam will add that in